### ISSUE TITLE
False-Positive "Function result is unused" with constructor calls

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1073]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.9 |
| Language level | PHP 7.2                        |

Current behaviour:
---

Here is some example code:

```php
<?php

declare(strict_types=1);

class A
{
    /** @var string */
    private $foo;

    public function __construct(string $foo)
    {
        $this->foo = $foo;
    }
}

class B extends A
{
    /** @var string */
    private $bar;

    public function __construct(string $foo, string $bar)
    {
        parent::__construct($foo);
        $this->bar = $bar;
    }
}

class C extends B
{
    /** @var string */
    private $fooBar;

    public function __construct(string $foo, string $bar, string $fooBar)
    {
        parent::__construct($foo, $bar);
        $this->fooBar = $fooBar;
    }
}
```

I get a warning "Function result is not used" in class C in the line `parent::__construct($foo, $bar);` now. The weird thing is, if class C extends from class A directly instead of class B, this false-positive does not happen.

Expected behaviour:
---

Do not report this. The constructor can't return anything.

**Comment by kalessil** [comment]
+ fluid interfaces (returning self, static or this)

**Comment by kalessil** [comment]
Fixed!

**Comment by samdark** [comment]
@kalessil was it released for EA Ultimate?

**Comment by samdark** [comment]
False positive is still there for me in the current version. It's reporting that constructor return result isn't used.
